### PR TITLE
[IMP] im_livechat: display country name with a known partner

### PR DESCRIPTION
**Pull request by kma-odoo**
Task Link: https://www.odoo.com/web#id=1940407&action=333&active_id=131&model=project.task&view_type=form&menu_id=4720

Pad Link: https://pad.odoo.com/p/r.9d900ba808bb10c651981946a52c50de

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/js/thread_windows/abstract_thread_window.js [hunk 1]
```diff
@@ -194,6 +194,17 @@ var AbstractThreadWindow = Widget.extend({
...
         return this._thread.getTitle();
     },
+    /**
+     * Get the country name of the direct partner.
+     *
+     * @returns {string}
+     */
+    getCountryName: function () {
+        if (!this.hasThread()) {
+            return _t("Undefined");
```
Comment (hmo-odoo) [inline addons/mail/static/src/js/thread_windows/abstract_thread_window.js]: should not display any string if country is not identified

File: addons/im_livechat/models/im_livechat_channel.py [hunk 2]
```diff
@@ -162,8 +162,10 @@ def _get_mail_channel(self, anonymous_name, previous_operator_id=None, user_id=N
         operator_partner_id = operator.partner_id.id
         # partner to add to the mail.channel
         channel_partner_to_add = [(4, operator_partner_id)]
-        if self.env.user and self.env.user.active:  # valid session user (not public)
-            channel_partner_to_add.append((4, self.env.user.partner_id.id))
+        if user_id:
+            current_user = self.env['res.users'].browse(user_id)
```
Comment (alexkuhn) [inline addons/im_livechat/models/im_livechat_channel.py]: This is visitor's user, not necessarily current user, hence I would name it `visitor_user` instead.

Comment (kma-odoo) [reply addons/im_livechat/models/im_livechat_channel.py]: @alexkuhn done, I've changed it to `visitor_user`

File: addons/im_livechat/models/mail_channel.py [hunk 3]
```diff
@@ -105,7 +105,14 @@ def _channel_get_livechat_partner_name(self):
...
             partners = self.channel_partner_ids - self.livechat_operator_id
             if partners:
-                return ', '.join(partners.mapped('name'))
+                partner_name = False
+                for partner in partners:
+                    if not partner_name:
+                        partner_name = partner.name
+                    else:
+                        partner_name += ', %s' % partner.name
+                    partner_name += ' (%s)' % partner.country_id.name if partner.country_id else ''
```
Comment (mge-odoo) [inline addons/im_livechat/models/mail_channel.py]: Is it possible to add a test for this?

Comment (mge-odoo) [reply addons/im_livechat/models/mail_channel.py]: I would use a `if` for better readability, like:

```
if partner.country_id:
    partner_name += ' (%s)' % parter.country_id.name

Comment (kma-odoo) [reply addons/im_livechat/models/mail_channel.py]: Done, would you please review it again?
I've added the test in https://github.com/odoo/odoo/pull/33831.
Thanks

**Comment by mge-odoo** [comment]
robodoo r+

**Pull request closed without merging**